### DISCUSSION TITLE
"Re-run failed jobs" option disappears if using re-usable workflows

### DISCUSSION THREAD

**Discussion by JackCoplandGranta** [community/community#13501] [Actions] [answered]
I was happy to see the "Re-run failed jobs" option [get added recently](https://github.blog/changelog/2022-03-17-github-actions-re-run-only-failed-or-individual-jobs/). We've got very long build chains and this will be a very useful feature. Unfortunately there seems to be a (undocumented?) restriction on that feature. If your workflow utilizes [reusable workflows](https://docs.github.com/en/actions/using-workflows/reusing-workflows) then that option disappears.

The option even disappears when the reusable workflow job is unrelated to other jobs in your workflow. If you have a workflow with job B which depends ("needs:") on job A (both not reusable workflows) you can perform partial re-runs fine if A passes and B fails. If you introduce a third job which is a reusable workflow (not dependent on, or depended on by, A or B) the option just disappears.

Is this restriction expected to be removed in future so that reusable workflows and the re-run failed option will work together? This is quite heavily impacting how we can use this feature.

**Comment by chrispat** [comment]
Yes we do plan to remove this restriction in the future.  The issues stems from they way the graph of jobs is calculated and persisted, the model is slightly different when you have reusable workflows.   The team is working on a fix and we hope to have it out in the next couple of weeks.

**Reply by chenghuang-mdsol** [reply]
Do you have a ticket that we can track?

**Reply by chrispat** [reply]
Not a public one.

**Reply by domgom** [reply]
I can't see the rerun option in my reusable workflow

**Comment by chrispat** [comment, marked as answer]
The fix for this has been deployed.

**Comment by thapaashesh** [comment]
All,

I used to have a 'Re-run all jobs' button in my GitHub runner, but it seems to have disappeared. I would appreciate any help in finding out how to get this button back.

**Reply by DefaultRyan** [reply]
Agreed. This used to work recently, but the UI for rerunning jobs has disappeared.
